internal/ocm/client_test.go

```diff
@@ -0,0 +1,123 @@
+package ocm
+
+import (
+	"context"
+	"fmt"
+	"io/ioutil"
+	"net/http"
+	"net/http/httptest"
+	"strings"
+	"testing"
+
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+)
+
+func TestClientDo_Success(t *testing.T) {
+	var (
+		recordedHttpRequest *http.Request
+		recordedBody        []byte
+	)
+	s := httptest.NewServer(http.HandlerFunc(func(rw http.ResponseWriter, r *http.Request) {
+		recordedHttpRequest = r
+		recordedBody, _ = ioutil.ReadAll(recordedHttpRequest.Body)
+		fmt.Fprintln(rw, `{"response":"works!"}`)
+	}))
+	defer s.Close()
```

Comment:
I think it makes sense to move the test server into the subtest function, because recorded request and body are  local to a specific subtest.